drivers/sensor/lis2mdl/Kconfig

```diff
@@ -11,6 +11,20 @@ menuconfig LIS2MDL
 
 if LIS2MDL
 
+choice LIS2MDL_OPERATION_MODE
+	prompt "Operation mode"
+	default LIS2MDL_CONTINUOUS_MODE
+	help
+	  Specify the type of operation mode to be used by driver.
+
+config LIS2MDL_SINGLE_MODE
+	bool "Single mode"
+
+config LIS2MDL_CONTINUOUS_MODE
```

Comment:
I think such setting should be made part of DeviceTree configuration. In case multiple instances are to be supported, only a single operation mode would be supported.

Reply:
I felt the same yesterday. Thanks to highllight this.